org/mozilla/jss/util/NativeProxy.java

```diff
@@ -51,10 +53,13 @@ public NativeProxy(byte[] pointer) {
      */
     protected NativeProxy(byte[] pointer, boolean track) {
         mPointer = pointer;
+        mHashCode = registryIndex.getAndIncrement();
+        if (mPointer != null) {
+            mHashCode = Objects.hash(mHashCode, Arrays.hashCode(mPointer));
```

Comment:
@edewata  -- I think this might be better and provide more random distribution of `mHashCode`.